lib/ip_blocklist_manager.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This module is used to manage multiple ip blocklists. It has the
+ * ability to initialize, load and reload itself on any list changes.
+ *
+ */
+var Promise = require('bluebird')
+
+module.exports = function (log, config) {
+  var IPBlocklist = require('./ip_blocklist')(log, config)
+
+  function IPBlocklistManager() {
+    this.ipBlocklists = []
+  }
+
+  IPBlocklistManager.prototype.load = function (lists) {
+    var self = this
+
+    if (!Array.isArray(lists)) {
+      throw Error('lists must be an array')
+    }
+
+    // Initialize and load a blocklist for each file path
+    var loadedLists = lists.map(function (listPath) {
+      var blocklist = new IPBlocklist()
+      self.ipBlocklists.push(blocklist)
+      return blocklist.load(listPath)
+    })
+
+    return Promise.all(loadedLists)
+  }
+
+  IPBlocklistManager.prototype.clear = function () {
+    this.ipBlocklists = []
+  }
+
+  IPBlocklistManager.prototype.contains = function (ipAddress) {
+    var self = this
+    var startTime = Date.now()
+    var endTime
+    var found = false
+
+    // Go through all blocklists and check if there is a match on the
+    // ipAddress.
+    self.ipBlocklists.forEach(function (blocklist) {
```

Comment:
`forEach` will continue the loop and call `blocklist.contains` on every list even once a match is found. I'd use [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) so that once a block is found, the lookup can stop right away.

```js
var found = self.ipBlocklists.some(function(list) { return list.contains(ipAddress) })
```